device-api/src/config/mod.rs

```diff
@@ -55,16 +55,16 @@ use crate::{Arch, DeviceError};
 /// use furiosa_device::DeviceConfig;
 ///
 /// // Using specific device names
-/// DeviceConfig::from_str("0:0"); // npu0pe0
-/// DeviceConfig::from_str("0:0-1"); // npu0pe0-1
+/// DeviceConfig::from_str("0:0").unwrap(); // npu0pe0
+/// DeviceConfig::from_str("0:0-1").unwrap(); // npu0pe0-1
 ///
 /// // Using device configs
-/// DeviceConfig::from_str("warboy*2"); // single pe x 2 (equivalent to "warboy(1)*2")
-/// DeviceConfig::from_str("warboy(1)*2"); // single pe x 2
-/// DeviceConfig::from_str("warboy(2)*2"); // 2-pe fusioned x 2
+/// DeviceConfig::from_str("warboy*2").unwrap(); // single pe x 2 (equivalent to "warboy(1)*2")
+/// DeviceConfig::from_str("warboy(1)*2").unwrap(); // single pe x 2
+/// DeviceConfig::from_str("warboy(2)*2").unwrap(); // 2-pe fusioned x 2
 ///
 /// // Combine multiple representations separated by commas
-/// DeviceConfig::from_str("0:0-1, 1:0-1"); // npu0pe0-1, npu1pe0-1
+/// DeviceConfig::from_str("0:0-1,1:0-1").unwrap(); // npu0pe0-1, npu1pe0-1
```

Comment:
It's great for testing purposes, but I'm not sure it's meaningful enough as an example code.
This may be an unnecessary concern, as users may assume that it's okay to use `unwrap`.
How about taking the result of `DeviceConfig::from_str` as an `if let` and validating it with an `assert`?

Reply:
Writing a `if let` phrase would be worse if we want to keep this example intuitive. I can also write as:
```
/// # let config = 
/// DeviceConfig::from_str("0:0"); // npu0pe0
/// # assert!(config.is_ok());
```
which is a little more complicated, but we can hide the assert from the reader. Will that be ok?

Reply:
I don't mind using unwrap either.

By the way, I think it would be better if the example showed what exception handling is required when the user provides invalid input.
But this is also optional. It's not required. I think the code in the PR is good as it is. This can be improved later along with the other parts.